username_api.py

```diff
@@ -12,11 +14,55 @@
 
 patterns = yaml.load(open('websites.yml'))
 
-def check_username(website, username):
-	url = patterns['urls'].get(website, 'https://{w}.com/{u}').format(
+def get_profile_url(website, username):
+	return patterns['urls'].get(website, 'https://{w}.com/{u}').format(
 		w=website,
-		u=username
-	)
+		u=username)
+
+def get_avatar(website, username):
+	data = patterns['avatar'][website]
+
+	if not data:
+		return None
+
+	url = get_profile_url(website, username)
+	if 'url' in data:
+		url = data['url'].format(u=username)
+
+	response = r.get(url)
+	if response.status_code == 404:
+		return None
+
+	soup = BeautifulSoup(response.text, 'html.parser')
```

Comment:
this should be inside the relevant branch of the `if` ; it is an expensive object to create